presto-orc/src/main/java/com/facebook/presto/orc/writer/FlatMapColumnWriter.java

```diff
@@ -0,0 +1,379 @@
+import com.facebook.presto.orc.metadata.statistics.ColumnStatistics;
+import com.facebook.presto.orc.metadata.statistics.IntegerStatisticsBuilder;
+import com.facebook.presto.orc.metadata.statistics.StatisticsBuilder;
+import com.facebook.presto.orc.metadata.statistics.StringStatisticsBuilder;
+import com.facebook.presto.orc.stream.LongOutputStream;
+import com.facebook.presto.orc.stream.PresentOutputStream;
+import com.facebook.presto.orc.stream.StreamDataOutput;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slice;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.facebook.presto.common.block.ColumnarMap.toColumnarMap;
+import static com.facebook.presto.orc.OrcEncoding.DWRF;
+import static com.facebook.presto.orc.metadata.ColumnEncoding.ColumnEncodingKind.DIRECT;
+import static com.facebook.presto.orc.metadata.ColumnEncoding.ColumnEncodingKind.DWRF_MAP_FLAT;
+import static com.facebook.presto.orc.metadata.CompressionKind.NONE;
+import static com.facebook.presto.orc.metadata.OrcType.OrcTypeKind.MAP;
+import static com.facebook.presto.orc.stream.LongOutputStream.createLengthOutputStream;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Math.toIntExact;
+import static java.util.stream.Collectors.toList;
+
+public class FlatMapColumnWriter
+        implements ColumnWriter
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(FlatMapColumnWriter.class).instanceSize();
+
+    /* Column Ids for the the Map, Key and Value */
+    private final int column;
+    private final int sequence;
+    private final int keyColumnId;
+    private final int valueColumnId;
+    /* Type data for for the Map */
+    private final MapType type;
+    private final Type keyType;
+    private final Type valueType;
+    /* We need the OrcType of the key to perform Type specific Block writes */
+    private final OrcType keyOrcType;
+    private final OrcType valueOrcType;
+    private final boolean compressed;
+    /* Column Encoding data for the Map */
+    private final ColumnEncoding columnEncoding;
+    private final ColumnEncoding keyColumnEncoding;
+    private final ColumnEncoding valueColumnEncoding;
+    private final Optional<DwrfDataEncryptor> dwrfEncryptor;
+    /* Metadata writer for the Map type and copy of the passed metadata writer to pass to FlatMapKeyNode writer */
+    private final CompressedMetadataWriter metadataWriter;
+    private final MetadataWriter inputMetadataWriter;
+    /* valueWriterFactory to generate streams for individual keys */
+    private final DwrfSequenceColumnWriterFactory valueWriterFactory;
+    private final int stringStatisticsLimitInBytes;
```

Comment:
why is this copying the stringStatisticsLimintInBytes ?

Reply:
We have to supply the statistics builder and for string statistics builder one of the parameters is stringStatisticsLimitInBytes